### DISCUSSION TITLE
Add user interface to edit part of yaml code

### DISCUSSION THREAD

**Discussion by cyprien-pixngraph** [microsoft/vscode-discussions#803] [Extension Development QnA] [answered]
Hello,
I'm using ESPHome (https://esphome.io/) with childrens. ESPHome have a web interface that is using Monaco editor to edit Yaml code.
Editor have powerful functionalities like autocompletion, link to documentation, etc. But for childrens, it's sometimes difficult to use...
So, it could be great to have a user interface that could "edit" each components (first level of yaml).
It could work like that 

- when i'm hover a component, a button is shown in the margin
- when i click on the button, a popup appearing with all the options of the component
- i change or add options
- when i close the popup, the yaml code is updated

I just would like to know if, for another VSCode project, there is a functionality  nearly like that.
Thanks in advance !

**Comment by alefragnani** [comment, marked as answer]
Hi @cyprien-pixngraph ,

You should be probably need use WebView (https://code.visualstudio.com/api/extension-guides/webview) on your extension. The technology you would use to allow a WYSIWYG / two way editing experience, would depend on your requirements.

I remember seeing TMS Web Core extension (https://marketplace.visualstudio.com/items?itemName=tmssoftware.tmswebcore) with an incredible UX. I'm not sure, but if I remember correctly, its webview was developed using React. Even tho, as you can see, it is doable.

Hope this helps

**Reply by github-cyprien** [reply]
Perfect ! Thank you very much for that answer !!
Hope it could work also in the monaco editor, i will look at it further...
